gcs/src/test/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemTest.java

```diff
@@ -266,4 +321,68 @@ public void testInitializeCompatibleWithHadoopCredentialProvider() {}
 
   @Override
   public void testRenameHnBucket() {}
+
+  private class InMemoryGoogleCloudStorageForRead extends InMemoryGoogleCloudStorage {
```

Comment:
nit: InMemoryGoogleCloudStorageForRead -> CustomInMemoryCloudStorage. 

Also, provide the description about what it is suppose to do.